main.nf

```diff
@@ -191,67 +235,655 @@ process get_software_versions {
     """
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
-    fastqc --version > v_fastqc.txt
-    multiqc --version > v_multiqc.txt
     scrape_software_versions.py > software_versions_mqc.yaml
     """
 }
 
+if (params.mzmlPaths) {
+  // Profile 'test' delivers mzmlPaths
+  Channel
+    .from(params.mzmlPaths)
+    .set { mzml_in }
+}
+else if (!params.mzmldef) {
+  Channel
+    .fromPath(params.mzmls)
+    .map { it -> [it, 'NA'] }
+    .set { mzml_in }
+} else {
+  Channel
+    .from(file("${params.mzmldef}").readLines())
+    .map { it -> it.tokenize('\t') }
+    .set { mzml_in }
+}
+
+
+// Parse mzML input to get files and sample names etc
+// get setname, sample name (baseName), input mzML file. 
+// Set platename to samplename if not specified. 
+// Set fraction name to NA if not specified
+mzml_in
+  .map { it -> [it[1], file(it[0]).baseName.replaceFirst(/.*\/(\S+)\.mzML/, "\$1"), file(it[0]), it[2] ? it[2] : it[1], it[3] ? it[3] : 'NA' ]}
+  .tap{ sets; strips; mzmlfiles; mzml_isobaric; mzml_quant; mzml_msgf }
+  .count()
+  .set{ amount_mzml }
+
+// Set names are first item in input lists, collect them for PSM tables and QC purposes
+sets
+  .map{ it -> it[0] }
+  .unique()
+  .tap { setnames_psm } 
+  .collect()
+  .map { it -> [it] }
+  .into { setnames_featqc; setnames_psmqc }
+
+// Strip names for HiRIEF fractionation are third item, 
+strips
+  .map { it -> it[3] }
+  .unique()
+  .toList()
+  .set { strips_for_deltapi }
 
 
 /*
- * STEP 1 - FastQC
- */
-process fastqc {
-    tag "$name"
-    publishDir "${params.outdir}/fastqc", mode: 'copy',
-        saveAs: {filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename"}
+* Step 1: Extract quant data from peptide spectra
+*/
 
-    input:
-    set val(name), file(reads) from read_files_fastqc
+process quantifySpectra {
+  when: !params.quantlookup && !params.noquant
+
+  input:
+  set val(setname), val(sample), file(infile), val(platename), val(fraction) from mzml_quant
+  file(hkconf) from Channel.fromPath("$baseDir/assets/hardklor.conf").first()
+
+  output:
+  set val(sample), file("${sample}.kr"), file(infile) into kronik_out
+  set val(sample), file("${infile}.consensusXML") optional true into isobaricxml
+
+  script:
+  activationtype = [hcd:'High-energy collision-induced dissociation', cid:'Collision-induced dissociation', etd:'Electron transfer dissociation'][params.activation]
+  massshift = [tmt:0.0013, itraq:0.00125, false:0][plextype]
+  """
+  cp $hkconf config
```

Comment:
This looks a bit weird to me: The $hkconf is already staged by Nextflow in the process specific work directory - no need to copy anything here...

If you need the content in a separate file, you could still do a simple `cat $hkconfig > config` here...